Bal 3307 - Improve Monthly Visitors Chart

shanegrouber opened the pull request:
- Add ChartContainer component for chart rendering
- Implement ChartTooltip and ChartTooltipContent components
- Enhance the WebsiteCredibility component to use the new Chart components

(your JSX structure is more nested than a Russian doll collection)

shanegrouber committed:
feat(chart): introduce ChartContainer and tooltip components

- Add ChartContainer component for chart rendering
- Implement ChartTooltip and ChartTooltipContent components
- Enhance the WebsiteCredibility component to use the new Chart components

(your JSX structure is more nested than a Russian doll collection)

tomer-shvadron commented on packages/ui/src/components/templates/report/components/WebsiteCredibility/WebsiteCredibility.tsx:
```diff
@@ -80,6 +84,19 @@ export const WebsiteCredibility: FunctionComponent<{
     [trafficAnalysis.trafficSources],
   );
 
+  const calculateTrend = (data: Array<{ label: string; value: string }>) => {
+    if (data.length < 2) return { direction: 'No trend data', percentage: 0 };
+    const lastMonthValue = parseInt(data[data.length - 1]?.value ?? '0');
+    const previousMonthValue = parseInt(data[data.length - 2]?.value ?? '0');
+    const percentageChange = ((lastMonthValue - previousMonthValue) / previousMonthValue) * 100;
+    const direction = lastMonthValue > previousMonthValue ? 'up' : 'down';
+    return { direction, percentage: Math.abs(percentageChange) };
+  };
```
```suggestion
  const calculateTrend = (data: Array<{ label: string; value: string }>) => {
    if (data.length < 2) {
      return { direction: 'No trend data', percentage: 0 };
    }
    
    const lastMonthValue = parseInt(data[data.length - 1]?.value ?? '0');
    const previousMonthValue = parseInt(data[data.length - 2]?.value ?? '0');
    const percentageChange = ((lastMonthValue - previousMonthValue) / previousMonthValue) * 100;
    const direction = lastMonthValue > previousMonthValue ? 'up' : 'down';
    
    return { direction, percentage: Math.abs(percentageChange) };
  };
```

shanegrouber committed:
fix(WebsiteCredibility): improve trend calculation readability

- Refactor conditional return for better clarity
- Enhance code structure for maintainability

(your code's readability is like a secret menu, good luck figuring it out)
